test/unit/toggle_cancel.ts

```diff
@@ -0,0 +1,91 @@
+import { expect } from "chai";
+import * as todoer from "../../src/core/todoer";
+
+describe("Cancelling a task", () => {
+  describe("Should be able to cancel an incomplete task", () => {
+    // prettier-ignore
+    const theories = [
+      [ "- [ ] Go shopping", "- [ ] ~~Go shopping~~" ], 
+      [ "- [ ] ",            "- [ ] ~~~~"            ], 
```

Comment:
Didn't really like that cancelling a blank line results in ~~~~ , but the logic remained simpler that way.  Will probably fix/change it in the near future.

Reply:
Nevermind.  I changed this behaviour.  You can no longer cancel or complete a task with no text.